### DISCUSSION TITLE
Sonoff Mini R4 Extreme

### DISCUSSION THREAD

**Discussion by tukestolo** [arendst/Tasmota#18698] [Support request]
Hey guys I saw a lot of videos telling how to flash Sonoff Mini R4 Extreme but I was not able to do it with success. 
I saw the 3.3v it is the U2 pin. But when I connect everything I am not able to have power on the device. I also try on a different device and it is the same. I connect to the 220v and it works like a charm. 
Can someone help me figure out why it lacks power when connected to 3v? it is possible to use another pin?

**Comment by sfromis** [comment]
It is entirely normal that devices designed for mains operation will not work fully on 3.3V power input. It may help if the 3.3V power source has at least 500 mA, but still not how it is meant to work, and relays usually needs at least 5V, if not more.

**Reply by tukestolo** [reply]
> It is entirely normal that devices designed for mains operation will not work fully on 3.3V power input. It may help if the 3.3V power source has at least 500 mA, but still not how it is meant to work, and relays usually needs at least 5V, if not more.

I do not think that is entirely true. Normally when we want to flash it the esp32, will be active with the 3v power. In this casa I am not able to do it… I don’t understand why.

**Reply by barbudor** [reply]
As you don't give much details on how you are doing that and which kind of USB/serial you are using, our best guess is that you don't use an adapter that juice the ESP (and the board) enough

https://tasmota.github.io/docs/Getting-Started/#serial-programmer

You must use an adapter with it's own 3V3 regulator to feed the board, and not a tiny one.

**Reply by sfromis** [reply]
> I do not think that is entirely true.

From the experience of me and others, it **is** normal that ESP32-based IoT devices will not connect to WiFi with only power from regular USB/serial adapters, of course depending on specific hardware. And if you want relay operation, you rarely can do that with only 3.3V available.

**Reply by tukestolo** [reply]
> > I do not think that is entirely true.
> 
> From the experience of me and others, it **is** normal that ESP32-based IoT devices will not connect to WiFi with only power from regular USB/serial adapters, of course depending on specific hardware. And if you want relay operation, you rarely can do that with only 3.3V available.

Hm for me all sonoff devices flash and connects to Wi-Fi only with the 3v.
But that’s another thing, for this case with the 3v the leds should blink at least.

**Reply by tukestolo** [reply]
> As you don't give much details on how you are doing that and which kind of USB/serial you are using, our best guess is that you don't use an adapter that juice the ESP (and the board) enough
> 
> https://tasmota.github.io/docs/Getting-Started/#serial-programmer
> 
> You must use an adapter with it's own 3V3 regulator to feed the board, and not a tiny one.

Sorry for weak explanation. But yes Iam using that one. Is the same that I used to flash all others ones that I have :/

**Reply by Jason2866** [reply]
Please link to one of this videos you looked how to do. We will have a look. Without knowing what is done. No one can say it is done correct or wrong.

**Reply by tukestolo** [reply]
For exemple, this one 
https://www.youtube.com/watch?v=erlwl13Jw8A

**Comment by Jason2866** [comment]
Have you connected gnd to GPIO 0 before connecting the USB serial adapter?

**Reply by tukestolo** [reply]
hm that connection is when we press the button while we are connecting the USB serial adapter.
Normally even if we do not put the device in flash mode, the blue led should link when it has 3v power :/

**Comment by aug70** [comment]
Mini R4 extreme can NOT be flashed with Tasmota. I don't know how digiblur did it but it didn't work for me on all 3 devices that I tried. It just doesn't work.

**Reply by Noschvie** [reply]
Have you tried how NotEnoughTech have flashed the device? That worked for me.
